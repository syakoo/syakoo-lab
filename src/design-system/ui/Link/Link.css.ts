import { recipe } from "@vanilla-extract/recipes";

import { tokens } from "@/design-system/tokens";

export const linkStyle = recipe({
  base: {
    display: "inline-block",
    textDecoration: "none",
    color: "currentColor",
    ":hover": {
      textDecoration: "none",
      opacity: 0.8,
      transition: "0.2s transform",
      transform: "translateY(-1px)",
    },
  },

  variants: {
    colored: {
      true: {
        color: tokens.colors.accent.info.foreground,
      },
    },
    underlined: {
      true: {
        ":hover": {
          textDecoration: "underline",
        },
      },
    },
    noHovered: {
      true: {
        ":hover": {
          transform: "none",
        },
      },
    },
    display: {
      block: {
        display: "block",
      },
    },
  },
});
