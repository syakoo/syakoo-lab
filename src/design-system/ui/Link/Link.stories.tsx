import type { Meta } from "@storybook/react";

import { tokens } from "@/design-system/tokens";

import { Link } from ".";

const meta = {
  component: Link,
  tags: ["autodocs"],
} satisfies Meta<typeof Link>;

export default meta;

export const Default = () => {
  return <Link href="#">Default Link</Link>;
};

export const Variants = () => {
  return (
    <>
      <div>
        <Link colored href="#">
          Colored Link
        </Link>
      </div>
      <div>
        <Link href="#" underlined>
          Underlined Link
        </Link>
      </div>
      <div>
        <Link href="#" noHovered>
          no hovered Link
        </Link>
      </div>
      <div>
        <Link display="block" href="#">
          <div
            style={{
              border: `1px solid ${tokens.colors.brand.primary}`,
              borderRadius: tokens.radii[100],
            }}
          >
            Block Link
          </div>
        </Link>
      </div>
    </>
  );
};
