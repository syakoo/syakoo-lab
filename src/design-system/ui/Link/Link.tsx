import React from "react";
import { linkStyle } from "./Link.css";

type LinkProps = {
  href: string;
  children: React.ReactNode;
  colored?: boolean;
  underlined?: boolean;
  noHovered?: boolean;
};

export const Link: React.FC<LinkProps> = ({
  href,
  children,
  colored = false,
  underlined = false,
  noHovered = false,
}) => {
  return (
    <a
      href={href}
      className={`${linkStyle({ colored, underlined, noHovered })}`}
    >
      {children}
    </a>
  );
};
