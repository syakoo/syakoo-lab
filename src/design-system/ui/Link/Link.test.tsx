import { testStories } from "@/test/testStories";

import * as stories from "./Link.stories";

describe("Link", () => {
  testStories(stories);
});
