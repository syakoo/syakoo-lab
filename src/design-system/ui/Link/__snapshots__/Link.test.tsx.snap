// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders Default 1`] = `
<body>
  <div>
    <a
      class="Link_linkStyle__m4v9430"
      href="#"
    >
      Default Link
    </a>
  </div>
</body>
`;

exports[`Link renders Variants 1`] = `
<body>
  <div>
    <div>
      <a
        class="Link_linkStyle__m4v9430 Link_linkStyle_colored_true__m4v9431"
        href="#"
      >
        Colored Link
      </a>
    </div>
    <div>
      <a
        class="Link_linkStyle__m4v9430 Link_linkStyle_underlined_true__m4v9432"
        href="#"
      >
        Underlined Link
      </a>
    </div>
    <div>
      <a
        class="Link_linkStyle__m4v9430 Link_linkStyle_noHovered_true__m4v9433"
        href="#"
      >
        no hovered Link
      </a>
    </div>
  </div>
</body>
`;
